src/state/slices/addressBookSlice/selectors.ts

```diff
@@ -0,0 +1,143 @@
+import { CHAIN_NAMESPACE, fromAccountId, fromChainId } from '@shapeshiftoss/caip'
+import { matchSorter } from 'match-sorter'
+import { createSelector } from 'reselect'
+
+import { addressBookSlice } from './addressBookSlice'
+
+import { createDeepEqualOutputSelector } from '@/state/selector-utils'
+import {
+  selectAccountAddressParamFromFilter,
+  selectChainIdParamFromFilter,
+  selectSearchQueryFromFilter,
+} from '@/state/selectors'
+import { selectAccountIdsWithoutEvms, selectEvmAccountIds } from '@/state/slices/common-selectors'
+
+export const selectAddressBookEntries = createSelector(
+  [addressBookSlice.selectors.selectEntriesByAccountId],
+  byAccountId => Object.values(byAccountId),
+)
+
+export const selectAddressBookEntriesByChainId = createDeepEqualOutputSelector(
+  selectChainIdParamFromFilter,
+  selectAddressBookEntries,
+  (chainId, entries) => {
+    if (!chainId) return []
+
+    return entries.filter(entry => entry.chainId === chainId)
+  },
+)
+
+export const selectAddressBookEntriesBySearchQuery = createDeepEqualOutputSelector(
+  selectChainIdParamFromFilter,
+  selectSearchQueryFromFilter,
+  selectAddressBookEntriesByChainId,
+  (_, searchQuery, entries) => {
+    if (!searchQuery) return entries
+
+    const matchedEntries = matchSorter(entries, searchQuery, {
+      keys: [
+        { key: 'label', threshold: matchSorter.rankings.MATCHES },
+        { key: 'address', threshold: matchSorter.rankings.CONTAINS },
```

Comment:
q: shouldn't this be the other way around? i.e matches for address, contains for label?